TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -372,6 +391,42 @@ public override bool Visit(DataDefinition dataDefinition)
             return true;
         }
 
+        public override bool Visit(End end)
+        {
+            // Check end statement is aligned with the matching opening statement
+            if (_compilerOptions.CheckEndAlignment && end.CodeElement.Type != CodeElementType.SentenceEnd)
+            {
+                CodeElement parentCodeElement = end.Parent.CodeElement;
+                if (parentCodeElement?.IsInsideCopy() == false)
```

Comment:
In fact you must check parentCodeElement?.IsInsideCopy() == false && end.IsInsideCopy() == false.
Because the parentCodeElement can be outside a copy and the end element inside a copy.

Reply:
done
https://github.com/TypeCobolTeam/TypeCobol/pull/1485/commits/6524f9aacb8470770c099d55de273434d1c6017c